src/mudlet.cpp

```diff
@@ -4122,6 +4122,553 @@ void mudlet::playSound(const QString& s, int soundVolume)
+}
+
+bool mudlet::isMSPFileRelative(QString soundFileName)
+{
+    bool isFileRelative = false;
+
+    if (!QFileInfo(soundFileName).isRelative()) {
+        qWarning() << QStringLiteral(
+            "mudlet::isMSPFileRelative() WARNING - Attempt made to send an absolute path as a sound file name: %1.  Only relative paths are permitted.")
+                .arg(soundFileName);
+    } else {
+        isFileRelative = true;
+    }
+
+    return isFileRelative;
+}
+
+QDir mudlet::getMSPDir(QString absoluteFolderPath)
+{
+    QDir mspDir(absoluteFolderPath);
+
+    if (!mspDir.exists()) {
+        mspDir.mkdir(absoluteFolderPath);
+    }
+
+    return mspDir;
+}
+
+QStringList mudlet::parseMSPFileNameList(bool isSound, QString& soundFileName, const QString& soundType, QDir& dir)
+{
+    QStringList fileNameList;
+    QPointer<Host> pHost = getActiveHost();
+
+    if (pHost) {
+        // No more than one wildcard of each individual type
+        if ((soundFileName.contains('*') || soundFileName.contains('?')) && (soundFileName.count('*') < 2 && soundFileName.count('?') < 2)) {
+            dir.setNameFilters(QStringList() << soundFileName);
+            QStringList fileNames(dir.entryList(QDir::Files | QDir::Readable, QDir::Name));
+
+            for (auto& fileName : qAsConst(fileNames)) {
+                if (!soundType.isEmpty()) {
+                    fileNameList << QStringLiteral("%1/%2/%3").arg(mudlet::getMudletPath(mudlet::profileSoundsPath, pHost->getName()), soundType, fileName);
+                } else {
+                    fileNameList << QStringLiteral("%1/%2").arg(mudlet::getMudletPath(mudlet::profileSoundsPath, pHost->getName()), fileName);
+                }
+            }
+        } else {
+            if (!soundFileName.contains('.')) {
+                soundFileName.append((isSound ? ".wav" : ".mid"));
+            }
+
+            if (!soundType.isEmpty()) {
+                fileNameList << QStringLiteral("%1/%2/%3").arg(mudlet::getMudletPath(mudlet::profileSoundsPath, pHost->getName()), soundType, soundFileName);
+            } else {
+                fileNameList << QStringLiteral("%1/%2").arg(mudlet::getMudletPath(mudlet::profileSoundsPath, pHost->getName()), soundFileName);
+            }
+        }
+    }
+
+    return fileNameList;
+}
+
+QStringList mudlet::getMSPFileNameList(bool isSound, QString& soundFileName, const QString &soundType)
+{
+    QStringList fileNameList;
+    QPointer<Host> pHost = getActiveHost();
+
+    if (pHost) {
+        QString soundsPath = mudlet::getMudletPath(mudlet::profileSoundsPath, pHost->getName());
+        QDir soundDir = mudlet::self()->getMSPDir(soundsPath);
+
+        if (soundDir.isEmpty()) {
+            qWarning() << QStringLiteral(
+                "mudlet::getMSPFileNameList() WARNING - Attempt made to create a directory failed: %1")
+                    .arg(mudlet::getMudletPath(mudlet::profileSoundsPath, pHost->getName()));
+            return fileNameList;
+        } else {
```

Comment:
:information_source: CLAZY *else-after-return* anti-pattern here - as the `if` being `true` case always leaves the method at the `return`  there is no need to have the `else {<code>}` - instead just put `<code>` directly on line 4244 (with the `} else {` on line 4243 changed to just a `}` - with one less level of indentation...! *Like what is the case for the `if` on line 4248 and the code following on line 4255.* :grinning:

Reply:
Hah :) - Yes, I understand.  Good catch.  Resolving.